minio/minio
# ilm: Make per-tier stats available via admin-tier-info

## Description
This change includes per-tier stats like total bytes and number of versions of objects transitioned to a tier. These statistics are made available via `mc admin tier info`.

## Motivation and Context
With this change users can track objects transitioned to a tier at a per-tier level.

## How to test this PR?
1. Configure one or more remote tiers.
2. Setup ILM policies to transition object to these tiers.
3. Enable versioning on test buckets; to observe number of versions transitioned.
4. `mc admin tier info ALIAS` (mc compiled with https://github.com/minio/madmin-go/pull/43)

Sample output
```
      $ ./mc admin tier info minio-tier
      Tier Name  |API  |Type  |Usage    |Objects  |Versions  
      STANDARD   |s3   |hot   |130 MiB  |1        |1         
      S3TIER-1   |s3   |warm  |130 MiB  |1        |3         
      S3TIER-2   |s3   |warm  |2.8 KiB  |0        |2

      $ ./mc admin tier info minio-tier --json | jq . 
      {
        "status": "success",
        "tiers": [
          {
            "Name": "STANDARD",
            "Type": "internal",
            "Stats": {
              "totalSize": 136314880,
              "numVersions": 1,
              "numObjects": 1
            }
          },
          {
            "Name": "S3TIER-1",
            "Type": "s3",
            "Stats": {
              "totalSize": 136317777,
              "numVersions": 3,
              "numObjects": 1
            }
          },
          {
            "Name": "S3TIER-2",
            "Type": "s3",
            "Stats": {
              "totalSize": 2897,
              "numVersions": 2,
              "numObjects": 0
            }
          }
        ]
      }       
```

## Types of changes

## Checklist:
